### PR TITLE
[bento4] Fix install to add cmake config file

### PR DESCRIPTION
**Pull request by RamadanAhmed**

### CODE REVIEW

File: ports/bento4/fix-install-and-c4996-error.patch [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+ Bento4Config.cmake |  2 ++
+ CMakeLists.txt     | 32 ++++++++++++++++++++++++++++++--
+ 2 files changed, 32 insertions(+), 2 deletions(-)
+ create mode 100644 Bento4Config.cmake
+
+diff --git a/Bento4Config.cmake b/Bento4Config.cmake
+new file mode 100644
+index 0000000..dc07afc
+--- /dev/null
++++ b/Bento4Config.cmake
```
Comment (PhoebeHui) [inline ports/bento4/fix-install-and-c4996-error.patch]: Could you add Bento4Config.cmake to port/bento4, instead of patching it?

**Comment by PhoebeHui** [comment]
@RamadanAhmed , thanks for the PR!

Could you open an issue for fixing the uwp failures? so others may replicate this problem.

**Comment by RamadanAhmed** [comment]
> Could you open an issue for fixing the uwp failures? so others may replicate this problem.
add_definitions(-D_CRT_SECURE_NO_WARNINGS) already exists in the old patch(thanks to NancyLi1013) but I forget to add it.

**Review by PhoebeHui — APPROVED** [review]

**Comment by dan-shaw** [comment]
Can you bump the control file?

File: ports/bento4/fix-install-and-c4996-error.patch [hunk 2]
```diff
@@ -1,17 +1,29 @@
+From cfb574bc4f1c8cda9fd33216cab6c53f60e1d487 Mon Sep 17 00:00:00 2001
+From: Ramadan <ramadan.ahmedr93@gmail.com>
+Date: Sat, 30 Nov 2019 13:31:40 +0200
+Subject: [PATCH] Fix install
+
+---
+ CMakeLists.txt     | 32 ++++++++++++++++++++++++++++++--
+ 1 files changed, 32 insertions(+), 2 deletions(-)
+ create mode 100644 Bento4Config.cmake
+
```
Comment (dan-shaw) [inline ports/bento4/fix-install-and-c4996-error.patch]: Can we remove the above lines so that the patch file is minimized?

**Comment by PhoebeHui** [comment]
@RamadanAhmed, can you please address the review suggestions? Thanks!

**Review by PhoebeHui — APPROVED** [review]

File: ports/bento4/fix-install-and-c4996-error.patch [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
-@@ -35,6 +35,8 @@ endif()
+@@ -33,7 +33,9 @@ else()
+   set(AP4_SOURCES ${AP4_SOURCES} ${SOURCE_SYSTEM}/Posix/Ap4PosixRandom.cpp)
+ endif()
  
- add_library(ap4 STATIC ${AP4_SOURCES})
- 
-+add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+-add_library(ap4 STATIC ${AP4_SOURCES})
++add_library(Bento4 STATIC ${AP4_SOURCES})
```
Comment (ras0219-msft) [inline ports/bento4/fix-install-and-c4996-error.patch]: Why is it required to rename the binary away from upstream?

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/bento4/fix-install-and-c4996-error.patch (hunk 3 above)
Comment (RamadanAhmed) [reply ports/bento4/fix-install-and-c4996-error.patch]: To be able to use like 
`target_link_libraries(main PRIVATE Bento4::Bento4)`
not 
`target_link_libraries(main PRIVATE Bento4::ap4)`

Comment (ras0219-msft) [reply ports/bento4/fix-install-and-c4996-error.patch]: Vcpkg will correctly detect the `Bento4::ap4` name and display help for the user, so I don't see that as a problem. Additionally, renaming the binaries causes incompatibilities with existing buildsystems which might use functions like
```
find_library(BENTO4 NAMES ap4)
```

We prefer to use upstream names whenever possible.

**Comment by PhoebeHui** [comment]
Pinging @RamadanAhmed for response.

**Comment by RamadanAhmed** [comment]
@PhoebeHui Sorry for being late, I think now its ready to be merged

File: ports/bento4/Bento4Config.cmake [hunk 4]
```diff
@@ -0,0 +1,2 @@
+include(CMakeFindDependencyMacro)
+include("${CMAKE_CURRENT_LIST_DIR}/Bento4Targets.cmake")
```
Comment (PhoebeHui) [inline ports/bento4/Bento4Config.cmake:2]: Could you update ' Bento4Targets.cmake' to 'bento4Targets.cmake'?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@RamadanAhmed thanks for your updates!

Could you also update the version from  1.5.1-628 to 1.5.1-628-1 in CONTROL file?

**Comment by RamadanAhmed** [comment]
> @RamadanAhmed thanks for your updates!
> 
> Could you also update the version from 1.5.1-628 to 1.5.1-628-1 in CONTROL file?

it should be 1.6.0-634

**Comment by PhoebeHui** [comment]
@RamadanAhmed, thanks for the updates!

Could you update 'Bento4Targets.cmake' to 'bento4Targets.cmake'? since it's case sensitive on Linux?

**Comment by PhoebeHui** [comment]
@RamadanAhmed, could you adress the review suggestions? Thanks!
1. Bump the version in CONTROL file.
2.  Update 'Bento4Targets.cmake' to 'bento4Targets.cmake' in Bento4Config.cmake file.

**Comment by PhoebeHui** [comment]
Closing this PR since it seems that no progress is being made, since the repository has been removed, however the changes is useful for us, I will summit a seprate PR for this.

**Pull request closed without merging**